tests/upgrade/raycluster_sdk_upgrade_test.py

```diff
@@ -0,0 +1,165 @@
+
+
+class TestMnistJobSubmit:
+    def setup_method(self):
+        initialize_kubernetes_client(self)
+        self.namespace = namespace
+        self.cluster = get_cluster("mnist", self.namespace)
+        if not self.cluster:
+            raise RuntimeError("TestRayClusterUp needs to be run before this test")
+
+    def teardown_method(self):
+        delete_namespace(self)
+
+    def test_mnist_job_submission(self):
+        auth = TokenAuthentication(
+            token=run_oc_command(["whoami", "--show-token=true"]),
+            server=run_oc_command(["whoami", "--show-server=true"]),
+            skip_tls=True,
+        )
+        auth.login()
+
+        self.assert_jobsubmit_withoutLogin(self.cluster)
+        self.assert_jobsubmit_withlogin(self.cluster)
+        self.cluster.down()
+
+    # Assertions
+    def assert_jobsubmit_withoutLogin(self, cluster):
+        dashboard_url = cluster.cluster_dashboard_uri()
+        jobdata = {
+            "entrypoint": "python mnist.py",
+            "runtime_env": {
+                "working_dir": "./tests/e2e/",
+                "pip": "./tests/e2e/mnist_pip_requirements.txt",
+            },
+        }
+        try:
+            response = requests.post(
+                dashboard_url + "/api/jobs/", verify=False, json=jobdata
+            )
+            if response.status_code == 403:
+                assert True
+            else:
+                response.raise_for_status()
+                assert False
+
+        except Exception as e:
+            print(f"An unexpected error occurred. Error: {e}")
+            assert False
+
+    def assert_jobsubmit_withlogin(self, cluster):
+        auth_token = run_oc_command(["whoami", "--show-token=true"])
+        ray_dashboard = cluster.cluster_dashboard_uri()
+        header = {"Authorization": f"Bearer {auth_token}"}
+        client = RayJobClient(address=ray_dashboard, headers=header, verify=False)
+
+        # Submit the job
+        submission_id = client.submit_job(
+            entrypoint="python mnist.py",
+            runtime_env={
+                "working_dir": "./tests/e2e/",
+                "pip": "./tests/e2e/mnist_pip_requirements.txt",
+            },
+        )
+        print(f"Submitted job with ID: {submission_id}")
+        done = False
+        time = 0
+        timeout = 900
+        while not done:
+            status = client.get_job_status(submission_id)
+            if status.is_terminal():
+                break
+            if not done:
+                print(status)
+                if timeout and time >= timeout:
+                    raise TimeoutError(f"job has timed out after waiting {timeout}s")
+                sleep(5)
+                time += 5
+
+        logs = client.get_job_logs(submission_id)
+        print(logs)
+
+        self.assert_job_completion(status)
+
+        client.delete_job(submission_id)
+        cluster.down()
+
+    def assert_job_completion(self, status):
+        if status == "SUCCEEDED":
```

Comment:
Wouldn't it have more sense to assert directly content of `status`?

Reply:
I synced these changes from [SDK-auth test](https://github.com/project-codeflare/codeflare-sdk/commit/88b2518db3657411635e647301e59dfdd65d75a7#diff-0d12faeac6160d6a2a47811c5f68013d2f459bf90d61151091b291e6782f5200R48). I think there are explicit print statements to have job_completion status for which the condition is added. 
But, Yes right! Can assert status directly 👀
Thank you @sutaakar for reviewing!